### PR TITLE
tests: crypto: Enable testing on smaller devices

### PR DESCRIPTION
**Pull request by torsteingrindvik**
Ref.: NCSDK-3513 , NCSDK-5903

In order to fit FLASH and SRAM for small devices, split execution in groups:

- Hash
- ECC
- AES

Also, enable reducing resources usage by turning off large vectors (in terms of both # and size) and by skipping execution of certain tests (hash) even though hash support is compiled in (this saves resources by not having the hash test buffers).

"Regular" devices work the same way as before.

Example of running on an emulated 52811 can be seen in `testcase.yaml`, it generally goes like this:

`west build -b nrf52840dk_nrf52811  -- -DOVERLAY_CONFIG="overlay-small-devices/overlay-small.conf;overlay-small-devices/overlay-small-aes.conf"`

Small devices currently do not have the freedom to choose backend freely.
Oberon is used for hash and ECC, mbed TLS is used for AES (because of the smaller footprint).

Depends on nrfxlib PR: https://github.com/nrfconnect/sdk-nrfxlib/pull/256

### CODE REVIEW

File: tests/crypto/test_cases/CMakeLists.txt [hunk 1]
```diff
@@ -11,14 +11,15 @@ zephyr_linker_sources(RODATA
     ${CMAKE_CURRENT_SOURCE_DIR}/custom-rodata.ld
 )
 
+if(NOT CONFIG_CRYPTO_SKIP_HASH_TESTS)
```
Comment (tejlmand) [inline tests/crypto/test_cases/CMakeLists.txt:14]: Please indent the lines between this `if()` and the following `endif()`

File: tests/crypto/test_cases/CMakeLists.txt [hunk 2]
```diff
@@ -41,13 +42,16 @@ zephyr_sources_ifdef(CONFIG_MBEDTLS_ECDSA_C         test_vectors_ecdsa_random.c)
 zephyr_sources_ifdef(CONFIG_MBEDTLS_ECDSA_C         test_vectors_ecdsa_sign.c)
 zephyr_sources_ifdef(CONFIG_MBEDTLS_ECDSA_C         test_vectors_ecdsa_verify.c)
 zephyr_sources_ifdef(CONFIG_MBEDTLS_ECJPAKE_C       test_vectors_ecjpake.c)
+if(NOT CONFIG_CRYPTO_SKIP_HASH_TESTS)
```
Comment (tejlmand) [inline tests/crypto/test_cases/CMakeLists.txt:45]: Please indent the lines between this `if()` and the following `endif()`

File: tests/crypto/test_cases/CMakeLists.txt [hunk 3]
```diff
@@ -41,13 +42,16 @@ zephyr_sources_ifdef(CONFIG_MBEDTLS_ECDSA_C         test_vectors_ecdsa_random.c)
...
 zephyr_sources_ifdef(CONFIG_MBEDTLS_SHA256_C        test_vectors_hkdf.c)
 zephyr_sources_ifdef(CONFIG_MBEDTLS_SHA256_C        test_vectors_hmac.c)
 zephyr_sources_ifdef(CONFIG_MBEDTLS_SHA256_C        test_vectors_sha_256.c)
 zephyr_sources_ifdef(CONFIG_MBEDTLS_SHA512_C        test_vectors_sha_512.c)
+endif()
 
 if(CONFIG_MBEDTLS_CMAC_C OR CONFIG_MBEDTLS_CMAC_ALT)
```
Comment (tejlmand) [inline tests/crypto/test_cases/CMakeLists.txt:52]: Please indent the lines between this `if()` and the following `endif()`

**Review by tejlmand — CHANGES_REQUESTED** [review]
Please add a link to the dependent `nrfxlib` PR.

File: tests/crypto/test_cases/CMakeLists.txt (hunk 1 above)
Comment (torsteingrindvik) [reply tests/crypto/test_cases/CMakeLists.txt:14]: Indented

File: tests/crypto/test_cases/CMakeLists.txt (hunk 2 above)
Comment (torsteingrindvik) [reply tests/crypto/test_cases/CMakeLists.txt]: Indented

File: tests/crypto/test_cases/CMakeLists.txt (hunk 3 above)
Comment (torsteingrindvik) [reply tests/crypto/test_cases/CMakeLists.txt]: Indented

**Review by frkv — APPROVED** [review]
LGTM

**Comment by carlescufi** [comment]
Please use a fork

**Pull request closed without merging**